x-pack/plugins/security_solution/server/lib/detection_engine/signals/filter_events_with_list.ts

```diff
@@ -34,63 +32,69 @@ export const filterEventsAgainstList = async ({
     const isStringableType = (val: SearchTypes) =>
       ['string', 'number', 'boolean'].includes(typeof val);
     // grab the signals with values found in the given exception lists.
-    const filteredHitsPromises = exceptionsList
-      .filter((exceptionItem: List) => exceptionItem.values_type === 'list')
-      .map(async (exceptionItem: List) => {
-        if (exceptionItem.values == null || exceptionItem.values.length === 0) {
-          throw new Error('Malformed exception list provided');
-        }
-        if (!type.is(exceptionItem.values[0].name)) {
-          throw new Error(
-            `Unsupported list type used, please use one of ${Object.keys(type.keys).join()}`
-          );
-        }
-        if (!exceptionItem.values[0].id) {
-          throw new Error(`Missing list id for exception on field ${exceptionItem.field}`);
-        }
-        // acquire the list values we are checking for.
-        const valuesOfGivenType = eventSearchResult.hits.hits.reduce((acc, searchResultItem) => {
-          const valueField = get(exceptionItem.field, searchResultItem._source);
-          if (valueField != null && isStringableType(valueField)) {
-            acc.add(valueField.toString());
-          }
-          return acc;
-        }, new Set<string>());
+    const filteredHitsPromises = exceptionsList.map(
+      async (exceptionItem: ExceptionListItemSchema) => {
+        const { entries } = exceptionItem;
 
-        // matched will contain any list items that matched with the
-        // values passed in from the Set.
-        const matchedListItems = await listClient.getListItemByValues({
-          listId: exceptionItem.values[0].id,
-          type: exceptionItem.values[0].name,
-          value: [...valuesOfGivenType],
-        });
+        const filteredHitsEntries = entries
+          .filter((t) => entriesList.is(t))
+          .map(async (entry) => {
+            // TODO: If this check is removed, then typescript doesn't
+            // recognize "entry" as EntryList despite .filter()
+            if (!entriesList.is(entry)) {
+              throw new Error('Malformed exception list provided');
+            }
```

Comment:
I think we can get rid of the check referenced by the TODO if we modify the filter a little bit 

```node.js
const filteredHitsEntries = entries
  .filter((t): t is EntryList => entriesList.is(t))

```
let me know if this does the trick.

Reply:
That seemed to do it, thanks!